iceoryx_posh/test/moduletests/test_popo_untyped_client.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/popo/untyped_client.hpp"
+#include "iceoryx_posh/testing/mocks/chunk_mock.hpp"
+#include "mocks/client_mock.hpp"
+
+#include "test.hpp"
+
+namespace
+{
+using namespace ::testing;
+using namespace iox::capro;
+using namespace iox::popo;
+using ::testing::_;
+
+using TestUntypedClient = iox::popo::UntypedClientImpl<MockBaseClient<void>>;
+
+class UntypedClient_test : public Test
+{
+  public:
+    void SetUp() override
+    {
+    }
+
+    void TearDown() override
+    {
+    }
+
+  protected:
+    ChunkMock<uint64_t, RequestHeader> requestMock;
+    ChunkMock<uint64_t, ResponseHeader> responseMock;
+
+    ServiceDescription sd{"oh", "captain", "my captain"};
+    static constexpr uint64_t RESPONSE_QUEUE_CAPACITY{123U};
+    ClientOptions options{RESPONSE_QUEUE_CAPACITY};
+    TestUntypedClient sut{sd, options};
+};
+
+TEST_F(UntypedClient_test, ConstructorForwardsArgumentsToBaseClient)
```

Comment:
Shouldn't we also verify:
* `getUid`
* `getServiceDescription`
* `connect`
* `getConnectionState`
* `disconnect`
* `hasResponse`
* `hasMissedResponse`
* `releaseQueuedResponse`

For instance the child could privatize or override the function with something else we do not want. Maybe some kind of typed test for the base{server,client} which the untyped/typed{server,client} has to pass?

Reply:
I don't think we do this for any derived class in out tests

Reply:
This is not correct, the functional interface will do this but for the moment you are safe ;)

Reply:
If you really really want, I can tackle this for the typed C++ API

Reply:
done in #1140